Examples/xforms-benchmark.lisp

```diff
@@ -0,0 +1,210 @@
+      (make-pane 'toggle-button :label "2" :id 2)
+      (make-pane 'toggle-button :label "3" :id 3)
+      (make-pane 'toggle-button :label "4" :id 4)
+      (make-pane 'toggle-button :label "5" :id 5)))
+   (recording
+    (with-radio-box ()
+      (radio-box-current-selection
+       (make-pane 'toggle-button :label "Off" :id 'off))
+      (make-pane 'toggle-button :label "On" :id 'on)))
+   (delay-finish
+    (with-radio-box ()
+      (radio-box-current-selection
+       (make-pane 'toggle-button :label "No delay" :id 0))
+      (make-pane 'toggle-button :label "0.25 sec." :id 0.25)
+      (make-pane 'toggle-button :label "1 sec." :id 1)
+      (make-pane 'toggle-button :label "2 sec." :id 2)))
+   (results :application
+            :min-width 600
+            :min-height 75
+            :end-of-line-action :scroll
+            :incremental-redisplay nil
+            :display-time t))
+  (:layouts
+   (default
+    (vertically ()
+      (horizontally ()
+        (labelling (:label "Operations") transform-ops)
+        (labelling (:label "Max Transforms") max-transforms)
+        (labelling (:label "Grid Size") grid-size)
+        (vertically ()
+          (labelling (:label "Output Recording") recording)
+          (labelling (:label "Delay FINISH-OUTPUT") delay-finish)))
+      canvas
+      (labelling (:label "Results") results)))))
+
+(defmethod run-xforms-benchmark (frame stream)
+  (let* ((results (find-pane-named frame 'results))
+         (ops (gadget-id (gadget-value (find-pane-named frame 'transform-ops))))
+         (max-transforms (gadget-id (gadget-value (find-pane-named frame 'max-transforms))))
+         (recording (gadget-id (gadget-value (find-pane-named frame 'recording))))
+         (delay-finish (gadget-id (gadget-value (find-pane-named frame 'delay-finish))))
+         (rows (gadget-id (gadget-value (find-pane-named frame 'grid-size))))
+                                        ; Number of rows of rectangles to display
+         (cols rows)                    ; Number of columns
+         (color-planes 8)               ; Number of different sets of colors
+         (inks (make-array (list color-planes cols rows)))
+         (itups internal-time-units-per-second))
+
+    (setf (stream-recording-p stream) (eq recording 'on))
+    (window-clear stream)
+    (window-clear results)
+
+    (dotimes (i color-planes)
+      (dotimes (j cols)
+        (dotimes (k rows)
+          (setf (aref inks i j k)
+                (make-rgb-color (float (/ (1+ i) color-planes))
+                                (float (/ (1+ j) cols))
+                                (float (/ (1+ k) rows)))))))
+
+    ;; XFORM-ITERATIONS should be an even number for each loop; this will
+    ;; prevent having an unmatched set of do/undo transformations.
+    (do* ((loop-cnt 0 (1+ loop-cnt))
+          (xform-iterations 0 (ash 1 (* loop-cnt 2)))
+          (s nil))
+
+        ((> xform-iterations max-transforms))
+
```

Comment:
there is no need for empty lines in the single top-level form (applies to other places too)